store/structured/mongo/store.go

```diff
@@ -45,21 +48,31 @@ func NewStore(p Params) (*Store, error) {
 		ctx:    context.Background(),
 	}
 
-	p.Lifecycle.Append(fx.Hook{
-		OnStart: func(ctx context.Context) error {
-			return store.Initialize(ctx)
-		},
-		OnStop: func(ctx context.Context) error {
-			return store.Terminate(ctx)
-		},
-	})
+	var err error
+
+	if p.Lifecycle != nil {
```

Comment:
I think this constructor should be split into two separate functions. One for creating the struct and a second one for attaching the lifecycle hooks. I gave you an example of how to do this in a comment in `store/structured/mongo/config.go`.